docs/architecture/dapr-for-net-developers/bindings-buildingblock.md

```diff
@@ -0,0 +1,235 @@
+4. After handling the event, the service returns an HTTP status code `200 OK`.
+
+The following ASP.NET Core controller provides an example of handling an event triggered by the Twitter binding:
+
+```csharp
+[ApiController]
+public class SomeController : ControllerBase
+{
+    [HttpPost("/tweet")]
+    public ActionResult Post(JsonElement data)
+    {
+        // handle tweet
+        //...
+
+        // acknowledge message
+        return Ok();
+    }
+  
+    //...  
+}
+```
+
+If the operation should error, you would return the appropriate 400 or 500 level HTTP status code. For bindings that feature *at-least-once-delivery* guarantees, the Dapr sidecar will retry the trigger. Check out [Dapr documentation for resource bindings][1] to see whether they offer *at-least-once* or *exactly-once* delivery guarantees.
+
+### Output bindings
+
+Dapr also includes *output binding* capabilities. They enable your service to trigger an event that invokes an external resource. Again, you start by configuring a binding configuration yaml file that describes the output binding. Once in place, you trigger an event that invokes the bindings API on the Dapr sidecar of your application. Figure 8-3 shows the architecture of an output binding:
+
+![Output binding](media/bindings-output.png)
+
+**Figure 8-3**. Dapr output binding.
+
+1. The Dapr sidecar reads the binding configuration file with the information on how to connect to the external resource. In the example, the external resource is a Twilio SMS account.
+2. Your application invokes the `/v1.0/bindings/sms` endpoint on the Dapr sidecar. In this case, it uses an HTTP POST to invoke the API. It's also possible to use gRPC.
+3. The binding component running in the Dapr sidecar calls the external messaging system to send the message. The message will contain the payload passed in the POST request.
+
+As an example, you can invoke an output binding by invoking the Dapr API using curl:
+
+```bash
+curl -X POST http://localhost:3500/v1.0/bindings/sms \
+  -H "Content-Type: application/json" \
+  -d '{
+        "data": "Welcome to this awesome service",
+        "metadata": {
+          "toNumber": "555-3277"
+        },
+        "operation": "create"
+      }'
+```
+
+Note that the HTTP port is the same as used by the Dapr sidecar (in this case, the default Dapr HTTP port `3500`).
+
+The structure of the payload (that is, message sent) will vary per binding. In the example above, the payload contains a `data` element with a message. Bindings to other types of external resources can be different, especially for the metadata that is sent. Each payload must also contain an `operation` field, that defines the operation the binding will execute. The above example specifies a  `create` operation that creates the SMS message. Common operations include:
+
+- create
+- get
+- delete
+- list
```

Comment:
What about `update`, is that a thing? I initially thought of CRUD when I read this list, but noticed that `update` isn't called out. Also, assuming that `create` is idempotent - it cannot be an update?

Reply:
Yes, that's possible. There is a default list of operation constants (see [definition](https://github.com/dapr/components-contrib/blob/aba184f9425b4e1950ad8d8e60d6912f37827d65/bindings/requests.go#L16)), but a binding can add its own operations to the list (in the end it's just a string).

I will add a remark about that in the text.